### PR TITLE
Fix spinlock in /nomarkshared view

### PR DESCRIPTION
**Pull request by mia-pi-git**

### CODE REVIEW

**Review by AnnikaCodes — CHANGES_REQUESTED** [review]

File: server/ip-tools.ts [hunk 1]
```diff
@@ -177,7 +177,10 @@ export const IPTools = new class {
 		const aParts = a.split('.');
 		const bParts = b.split('.');
 		while (diff === 0) {
-			diff = (parseInt(aParts[i]) || 0) - (parseInt(bParts[i]) || 0);
+			const aPart = parseInt(aParts[i]);
+			const bPart = parseInt(bParts[i]);
+			if (isNaN(aPart) || isNaN(bPart)) throw new Error("Invalid IP passed to IPTools.ipSort.");
+			diff = (aPart || 0) - (bPart || 0);
```
Comment (AnnikaCodes) [inline server/ip-tools.ts:183]:
```suggestion
			diff = aPart - bPart;
```
parseInt returns an integer or NaN, nothing falsy except 0

**Review by AnnikaCodes — CHANGES_REQUESTED** [review]

**Review by AnnikaCodes — APPROVED** [review]
Thanks!

File: server/chat-plugins/hosts.ts [hunk 2]
```diff
@@ -135,7 +135,7 @@ export const pages: PageTable = {
 		} else {
 			buf += `<div class="ladder"><table><tr><th>IP</th><th>Reason</th></tr>`;
 			const sortedSharedIPBlacklist = [...Punishments.sharedIpBlacklist];
-			sortedSharedIPBlacklist.sort((a, b) => IPTools.ipSort(a[1], b[1]));
+			sortedSharedIPBlacklist.sort((a, b) => IPTools.ipSort(a[0], b[0]));
```
Comment (Zarel) [inline server/chat-plugins/hosts.ts:138]: Should probably be `ipCompare`? Also I wonder if `sortBy` would be easier?

Comment (Zarel) [reply server/chat-plugins/hosts.ts:138]:
```ts
Utils.sortBy(sortedSharedIPBlacklist, ip => ip.split('.').map(octet => parseInt(octet)));
```

**Pull request merged**